test/code/plugins/npmd_config_lib_test.rb

```diff
@@ -16,57 +16,61 @@ class << self
-        @string_agent_config12='<Configuration><Agents><Agent Name="1" Capabilities="1"><IPConfiguration IP="65.171.126.72" Mask="255.255.0.0"/></Agent></Agents><Networks><Network Name="Default"><Subnet ID="65.171.0.0" Disabled="False" Tag=""/><Subnet ID="198.165.0.0" Disabled="False" Tag=""/></Network></Networks><Rules><Rule Name="Default" Description="" Protocol="1"><AlertConfiguration><Loss Threshold="-1.0"/><Latency Threshold="-1.0"/></AlertConfiguration><NetworkTestMatrix><SubnetPair SourceSubnet="*" SourceNetwork="*" DestSubnet="*" DestNetwork="*"/></NetworkTestMatrix><Exceptions /></Rule><Rule Name="RuleToPartialDrop" Description="" Protocol="1"><AlertConfiguration><Loss Threshold="-1.0"/><Latency Threshold="-1.0"/></AlertConfiguration><NetworkTestMatrix><SubnetPair SourceSubnet="65.171.0.0" SourceNetwork="Default" DestSubnet="198.165.0.0" DestNetwork="NewOne"/></NetworkTestMatrix><Exceptions /></Rule></Rules></Configuration>'.gsub(/\s+/,"")
+        #@test_desc01 = "Test Case 01: Valid UI XML Configuration with default rule"
+        #@test_desc02 = "Test Case 02: Invalid configuration version"
+        #@test_desc03 = "Test Case 03: Mismatched XML tags"
+        #@test_desc04 = "Test Case 04: Invalid configuration tags"
+        #@test_desc05 = "Test Case 05: Invalid json between tags"
+        #@test_desc06 = "Test Case 06: Missing agent capabilities"
+        #@test_desc07 = "Test Case 07: Missing rule protocol"
+        #@test_desc08 = "Test Case 08: Undefined subnet ids in config"
+        #@test_desc09 = "Test Case 09: IPV6 config gets compressed ips"
+        #@test_desc10 = "Test Case 10: Valid UI XML configuration with custom rule"
+        #@test_desc11 = "Test Case 11: Valid UI XML Configuration with exceptions"
+        #@test_desc12 = "Test Case 12: Verifying config error summary"
+        @test_desc13 = "Test Case 13: Verifying config error summary"
```

Comment:
i would suggest to test both config versions old and new after CMv2 changes.

Reply:
Updated config unit tests for both old and new configs.